src/main/kotlin/org/pkl/lsp/VirtualFileManager.kt

```diff
@@ -18,20 +18,9 @@ package org.pkl.lsp
 import java.net.URI
 import java.nio.file.*
 import java.util.*
-import java.util.concurrent.CompletableFuture
 import java.util.concurrent.ConcurrentHashMap
 
 class VirtualFileManager(project: Project) : Component(project) {
-  override fun initialize(): CompletableFuture<*> {
-    project.messageBus.subscribe(textDocumentTopic) { event ->
-      val file = files[event.file] ?: return@subscribe
-      if (event.type == TextDocumentEventType.CHANGED) {
-        file.modificationCount.incrementAndGet()
-      }
-    }
-    return CompletableFuture.completedFuture(Unit)
-  }
-
```

Comment:
I realized that the LSP tells us about the version of these files when they get opened; see https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentItem

So, I changed the TextDocumentService to just update `version` on each `VirtualFile`, and this version can be used as our modification count.